src/r_util.cpp

```diff
@@ -489,19 +491,41 @@ namespace rhost {
         }
 
         extern "C" SEXP fetch_file(SEXP path) {
-            fs::path fpath;
+            const char* f_path = nullptr;
+            
             r_top_level_exec([&]() {
-                fpath = fs::u8path(Rf_translateCharUTF8(STRING_ELT(path, 0)));
+                f_path = Rf_translateCharUTF8(STRING_ELT(path, 0));
             }, __FUNCTION__);
 
-            if (!fpath.empty()) {
-                blobs::blob file_data;
-                blobs::append_from_file(file_data, fpath);
-                host::send_notification("!FetchFile", file_data, fpath.filename().string());
-                return R_TrueValue;
-            }
-            
-            return R_FalseValue;
+            return util::exceptions_to_errors([&]() {
+                fs::path fpath = fs::u8path(f_path);
+                if (!fpath.empty()) {
+                    blobs::blob file_data;
+                    blobs::append_from_file(file_data, fpath);
+                    host::send_notification("!FetchFile", file_data, fpath.filename().string());
+                    return R_TrueValue;
+                }
+                return R_FalseValue;
+            });
+        }
+
+        extern "C" SEXP save_to_project_folder(SEXP id, SEXP project_name, SEXP dest_dir, SEXP temp_dir) {
+            auto blob_id = static_cast<blobs::blob_id>(Rf_asReal(id));
+            const char *prj_name = nullptr;
+            const char *t_dir = nullptr;
+            const char *d_dir = nullptr;
+
+            r_top_level_exec([&]() {
```

Comment:
>r_top_level_exec [](start = 12, length = 16)

No need for this anymore - its only function is to swallow R errors (and let you examine them at leisure), but in this case you actually want R errors to propagate out of this function to the caller, if any Rf_translateChar calls fail.